### ISSUE TITLE
rename module and create import hook

### ISSUE THREAD

**Issue by FlorianLudwig** [Yakifo/amqtt#4]
rename the module and create an import hook to allow importing the old name.

- [x] Decide on module name
- [x] rename repo
- [ ] rename package to match module name (blocked by https://github.com/Yakifo/amqtt/issues/7)
- [ ] Implement import hook & deprecation warning when importing the old name

**Comment by FlorianLudwig** [comment]
Possible names

 - `aio-hbmqttt`
 - `aiohbmqttt`
 - `aiomqttt` (taken)
 - `amqtt`

**Comment by HerrMuellerluedenscheid** [comment]
I like `amqtt`.

**Comment by FlorianLudwig** [comment]
Not to be confused with `amqp` :]

**Comment by FlorianLudwig** [comment]
Related discussion here: https://discuss.python.org/t/packaging-forks/7502

**Comment by FlorianLudwig** [comment]
We need to decide on an "official writing".
Currently both `aMQTT` and `AMQTT` are used.

@HerrMuellerluedenscheid your take?

**Comment by HerrMuellerluedenscheid** [comment]
> We need to decide on an "official writing".
> Currently both `aMQTT` and `AMQTT` are used.
> 
> @HerrMuellerluedenscheid your take?

Good point. I like `aMQTT`.

**Comment by FlorianLudwig** [comment]
Okay, lets go with `aMQTT`

**Comment by shipmints** [comment]
I'd suggest following the convention here https://github.com/aio-libs to indicate similar intent.

**Comment by FlorianLudwig** [comment]
@shipmints that is not possible as `aiomqtt` is already taken :(

**Comment by shipmints** [comment]
Perhaps, then, aiomqttd (in the Unix world, d implies server vs. client). That would suit me! If the person who owns aiomqtt as a client side implementation agrees, we could attempt to ensure compatibility.

**Comment by FlorianLudwig** [comment]
aiomqtt is a wrapper around paho, for details see: https://github.com/sbtinstruments/asyncio-mqtt/discussions/50

Since this repo does also provide a client implementation I would prefer to stay away from the daemon naming.

**Comment by HerrMuellerluedenscheid** [comment]
@FlorianLudwig I took a look at possibilities to setup import hooks and show deprecation warnings but then wasn't sure about the best strategy. I feel like this requires a deeper understanding of the `importlib` and import mechanism than I currently have. But I'm eager to understand how this should be done right.

You have something in mind like if I run
```
import hbmqtt
```
I see something like:
```
>>> UserWarning: HBMQTT is deprecated. Use aMQTT as drop-in replacement
```
right? `DeprecationWarning` or `PendingDeprecationWarning` would be nicer but those are ignored by default.

Well, I'm not sure how to approach the hooks. Wouldn't that require to register `aMQTT` and `hbmqtt` at installation e.g. by somehow hijacking `setup.py` logic? Do you happen to have some reference where I could have a look how that should be done?

**Comment by FlorianLudwig** [comment]
@HerrMuellerluedenscheid I also played around and my conclusion is the same, there needs to be some hack to make it work.

So my thought was to generate python files, "KISS":

for example the `hbmqtt/errors.py` would be:
```
import warnings
from amqtt.errors import *

warnings.warn("importing hbmqtt is deprecated. Please import amqtt", DeprecationWarning)
```

I made a small script to generate these files and works as expected.

The `DeprecationWarning` being ignored by default is something I thought about as well... I would use it in the next release and maybe use `FutureWarning` in a later release?
